### PR TITLE
[QuickOpen] style matched fuzzy characters (proper)

### PR DESCRIPTION
**Pull request by atwalg2**
Associated Issue: #4488

### Summary of Changes

*  Implement proper fuzzy highlighting based off fuzzaldrins-plus scoring scheme.
*  Added shadows to make highlights a little more pronounced

### Todo
- [x] Implement highlighting
- [x] handle modifiers ("@", ... ) in front of query
- [x] highlight titles when paths given
- [x] rewrite more efficiently, using .wrap()

### Screens

This screen shows off the highlighting shadows. 
image

image

image

image

### CODE REVIEW

File: src/utils/quick-open.js [hunk 1]
```diff
@@ -128,43 +128,44 @@ export function formatSources(sources: SourcesMap): Array<QuickOpenResult> {
...
-  });
+
+    first = last;
+    last = matches[counter] + 1;
+    output.push({
+      type: "match",
+      value: input.substr(first, last - first)
+    });
+
+    counter++;
```
Comment (jasonLaster) [inline src/utils/quick-open.js]: lets add some additional unit tests

File: src/components/QuickOpenModal.js [hunk 2]
```diff
@@ -3,6 +3,7 @@
  * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
 
 // @flow
+const path = require("path");
```
Comment (jasonLaster) [inline src/components/QuickOpenModal.js]: will this work w/ webpack?

File: src/utils/quick-open.js [hunk 3]
```diff
@@ -21,6 +21,11 @@ export const MODIFIERS = {
   "?": "shortcuts"
 };
 
+export const PATH = {
```
[the remark this answers is not in the document]
Comment (jasonLaster) [reply src/utils/quick-open.js]: do we have access to Node in the browser. We can add something to `utils/path`

File: src/components/QuickOpenModal.js (hunk 2 above)
Comment (atwalg2) [reply src/components/QuickOpenModal.js]: I'm not sure how to test this

Comment (wldcordeiro) [reply src/components/QuickOpenModal.js]: I think you should be able to do `import path from "path";` and yeah Webpack should do the work here if I recall.

**Comment by jasonLaster** [comment]
Pushed a couple of tweaks
+ an `[Object Object]` classname
+ dropped text-shadow. In the screenshot you'll see the title has it but the subtitle doesn't. in this case the subtitle looks sharper.

screen shot 2018-02-05 at 8 43 58 am
screen shot 2018-02-05 at 8 45 21 am

**Pull request merged**